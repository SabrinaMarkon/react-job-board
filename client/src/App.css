body {
  margin: 20px 40px;
  background: #535353;
  color: #fff;
}

#uopeoplelogo {
  max-width: 200px;
  float: right;
  margin-top: -70px;
  border: 0;
}

.jobandlogo {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
}

.jobbox {
  text-align: left;
}

.job {
  margin: 0 auto;
  display: flex;
  justify-content: space-between;
  margin: 20px 0;
  padding: 20px 10px;
  align-items: center;
  color: #393939;
}

.job:hover {
  cursor: pointer;
  color: #666666;
}

.job-company {
  padding: 0 30px;
}

.job-dialog {
  padding: 30px 50px;
}

#pagenum {
  margin-bottom: 1rem;
}

/* Footer */
.madewithlove {
  width: 100%;
  text-align: center;
  padding: 1rem;
  color: #fff;
}
.madewithlove a {
  color: #ff006a;
  text-decoration: none;
}
.madewithlove a:hover {
  color: #fff;
}
.madewithlove__heart {
  color: #ff006a;
}
.madewithlove__heart:hover {
  color: #fff;
}

@media(min-width: 200px) {
  .joblogo {
    max-width: 200px;
  }
}

@media(min-width: 700px) {
  .joblogo {
    max-width: 150px;
  }
}