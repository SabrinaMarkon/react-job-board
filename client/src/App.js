import React, { useState, useEffect } from 'react';
import './App.css';
import Jobs from './Jobs';

// We SEE the job site on port 3000 but it fetches the jobs from port 3001!
const JOB_API_URL = 'http://localhost:3001/jobs'; // api/index.js url! (local dev environment).
// const JOB_API_URL = 'http://sabrinamarkon.com:3001/jobs'; // api/index.js url!

async function fetchJobs(callbackSetJobsList) {
  const response = await fetch(JOB_API_URL);
  const json = await response.json();
  callbackSetJobsList(json);
}

// App component will hold state.
function App() {

  const [jobsList, setJobsList] = useState([]);

  useEffect(() => {
    fetchJobs(setJobsList).catch(e => 
      console.log(`Error with fetchJobs call: ${e}`));
  }, []);

  return (
    <div className="App">
      <Jobs jobs={jobsList} />
    </div>
  );
}

export default App;
