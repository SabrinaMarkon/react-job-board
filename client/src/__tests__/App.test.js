import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "../App";

afterEach(cleanup);

it("matches the App snapshot", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it("includes the h1 text", () => {
  const { getByText } = render(<App />);
  const h1Text = getByText(/Software Jobs for UoPeople Students and Alumni/);
  expect(h1Text).toBeInTheDocument();
});

it("includes the h2 text", () => {
  const { getByText } = render(<App />);
  const h2Text = getByText(/Found \d+ Jobs/);
  expect(h2Text).toBeInTheDocument();
});

it("includes the UoPeople logo", () => {
  const { getByAltText } = render(<App />);
  const uopeopleImg = getByAltText(/University of the People Computer Science/);
  expect(uopeopleImg).toBeInTheDocument();
});

it("Has the page number text for the pagination", () => {
  const { getByText } = render(<App />);
  const pageNumText = getByText(/Page \d+ of \d+/);
  expect(pageNumText).toBeInTheDocument();
});

it("renders the copyright link", () => {
  const { getByText } = render(<App />);
  const footerLink = getByText(/Sabrina Markon/);
  expect(footerLink).toBeInTheDocument();
});

it("renders the heart in the footer", () => {
  const { getByText } = render(<App />);
  const heartElement = getByText(/‚ù§/);
  expect(heartElement).toBeInTheDocument();
});
