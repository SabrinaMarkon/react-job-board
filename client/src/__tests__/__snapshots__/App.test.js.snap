// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the App snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="App"
    >
      <div />
      <h1
        class="MuiTypography-root MuiTypography-h4"
      >
        Software Jobs for UoPeople Students and Alumni
      </h1>
      <h2
        class="MuiTypography-root MuiTypography-h6"
      >
        Found 0 Jobs
      </h2>
      <a
        href="https://infl.tv/hvno"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="University of the People Computer Science"
          id="uopeoplelogo"
          src="logo_uopeople.png"
        />
      </a>
      <div
        id="pagenum"
      >
        Page 1 of 0
      </div>
      <div
        class="MuiPaper-root MuiMobileStepper-root MuiMobileStepper-positionStatic MuiPaper-elevation0"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
              />
            </svg>
            Back
          </span>
        </button>
        <div
          aria-valuenow="0"
          class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiMobileStepper-progress MuiLinearProgress-determinate"
          role="progressbar"
        >
          <div
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
            style="transform: translateX(-100%);"
          />
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Next
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <footer
        class="madewithlove"
      >
        ©2020 Made with 
        <span
          aria-hidden="true"
          class="madewithlove__heart"
        >
          ❤
        </span>
         for my fellow University of the People students by 
        <a
          class="madewithlove__name"
          href="http://sabrinamarkon.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          Sabrina Markon
        </a>
        , BSc Computer Science Graduate, 2019.
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
